crates/api/api/src/local_user/export_data.rs

```diff
@@ -0,0 +1,162 @@
+use activitypub_federation::config::Data;
+use actix_web::web::Json;
+use lemmy_api_utils::context::LemmyContext;
+use lemmy_db_schema::{
+  source::{
+    actor_language::LocalUserLanguage,
+    community::CommunityActions,
+    instance::InstanceActions,
+    keyword_block::LocalUserKeywordBlock,
+    person::PersonActions,
+  },
+  traits::Blockable,
+};
+use lemmy_db_views_community_follower::CommunityFollowerView;
+use lemmy_db_views_community_moderator::CommunityModeratorView;
+use lemmy_db_views_inbox_combined::{impls::InboxCombinedQuery, InboxCombinedView};
+use lemmy_db_views_local_user::LocalUserView;
+use lemmy_db_views_person_content_combined::{
+  impls::PersonContentCombinedQuery,
+  PersonContentCombinedView,
+};
+use lemmy_db_views_person_liked_combined::{
+  impls::PersonLikedCombinedQuery,
+  PersonLikedCombinedView,
+};
+use lemmy_db_views_person_saved_combined::{
+  impls::PersonSavedCombinedQuery,
+  PersonSavedCombinedView,
+};
+use lemmy_db_views_post::PostView;
+use lemmy_db_views_site::api::{ExportDataResponse, PostOrCommentOrPrivateMessage};
+use lemmy_utils::{self, error::LemmyResult};
+
+pub async fn export_data(
+  context: Data<LemmyContext>,
+  local_user_view: LocalUserView,
+) -> LemmyResult<Json<ExportDataResponse>> {
+  let local_user_id = local_user_view.local_user.id;
+  let local_instance_id = local_user_view.person.instance_id;
+  let my_person_id = local_user_view.person.id;
+  let my_person = &local_user_view.person;
+  let local_user = &local_user_view.local_user;
+
+  let pool = &mut context.pool();
+
+  let content = PersonContentCombinedQuery {
+    no_limit: Some(true),
+    ..PersonContentCombinedQuery::new(my_person_id)
+  }
+  .list(pool, Some(&local_user_view), local_instance_id)
+  .await?
+  .into_iter()
+  .map(|u| match u {
+    PersonContentCombinedView::Post(pv) => PostOrCommentOrPrivateMessage::Post(pv.post),
+    PersonContentCombinedView::Comment(cv) => PostOrCommentOrPrivateMessage::Comment(cv.comment),
+  })
+  .collect();
```

Comment:
These do mappings now. If they were used anywhere else it'd make sense to make a trait for this conversion, but its only used here.

Reply:
Add `use PostOrCommentOrPrivateMessage::*` inside the top of this function to cleanup the mappings.